vowpalwabbit/cs_active.cc

```diff
@@ -354,18 +354,36 @@ base_learner* cs_active_setup(VW::setup_base_i& stack_builder)
   all.set_minmax(all.sd, data->cost_min);
   for (uint32_t i = 0; i < data->num_classes + 1; i++) data->examples_by_queries.push_back(0);
 
-  learner<cs_active, example>& l = simulation
-      ? init_learner(data, as_singleline(stack_builder.setup_base_learner()), predict_or_learn<true, true>,
-            predict_or_learn<false, true>, data->num_classes, prediction_type_t::active_multiclass,
-            stack_builder.get_setupfn_name(cs_active_setup) + "-sim", true)
-      : init_learner(data, as_singleline(stack_builder.setup_base_learner()), predict_or_learn<true, false>,
-            predict_or_learn<false, false>, data->num_classes, prediction_type_t::active_multiclass,
-            stack_builder.get_setupfn_name(cs_active_setup), true);
+  void (*learn_ptr)(cs_active & cs_a, single_learner & base, example & ec);
+  void (*predict_ptr)(cs_active & cs_a, single_learner & base, example & ec);
+  size_t ws = data->num_classes;
```

Comment:
This doesn't need to be changed in the `if` statement, so you could pass it directly on line 377?

Reply:
I can move it after the `if/else` but it does need to be set before `make_reduction_learner` because of `std::move(data)`

Reply:
Oh I see, good point! Feel free to leave as is